### ISSUE TITLE
Can't upgrade forge on server (already submitted issue on atm10 github but no response)

### ISSUE THREAD

**Issue by AceForKnut** [stal111/Forbidden-Arcanus#450]
**Describe the bug**

**To Reproduce**
1. Go on a server
2. Use mundabitr dust
3. Try to upgrade forge

**Expected behavior**
For the forge to glow and be able to use the gavel on it

**Screenshots**
![image_2024-11-10_164543294](https://github.com/user-attachments/assets/d3f4b2d4-65cd-46d1-875d-78c6bea8be84)

**Versions**
 - Forbidden & Arcanus:
 - Valhelsia Core:
 - Forge:

**Additional context**
I'm playing on a server that regularly updates(atm10). The issue is that making the Forge (clicking it with mundabit dust) goes well, but the issue begins when you try to place down the pillars. Both types. the forge cant upgrade to a higher tier, and trying to make a new forge WHEN the pillars are present, won't work. Basically i believe it stems from the pillars not "cooperating" properly with the forge that's causing the bug is my theory. Image provided is the setup. the forge here IS tier 2 (cause an admin came and gave me one cause i had used items and he thought that would fix it. i have tried again from the bottom btw)

**Comment by AceForKnut** [comment]
Any information i can get i'll take ty

**Comment by Vanquished79** [comment]
I have same issue I'm able to upgrade to Tier 2 but nothing happens when trying to upgrade to Tier 3 I'm on a server 

I made sure all the required items were present but when trying to upgrade to Tier 3 nothing happens

**Comment by Erelaszun** [comment]
nothing in logs to submit... but I was having this same issue (not on server, on local Minecraft/Neoforge 1.21.1 Modpack I am building)

I also tested trying to make Ferrogneic Mixture, which only needs tier 1, also doesn't work... and same thing... if you remove the tier 1 forge, place down a smithing table, and try to use Mundabitur dust, it will not re-form a tier 1 until the pedestals are removed... so I agree 100% with the first post... its like the pedestals are not working with the rest of the structure.

**Comment by Erelaszun** [comment]
Just tried with the latest version, 2.5.10, and no change... I even completely tore down the structure and rebuilt it in a different area, and no change.

**Comment by itschammyg** [comment]
> Just tried with the latest version, 2.5.10, and no change... I even completely tore down the structure and rebuilt it in a different area, and no change.

Both the client and server are running 2.5.10? If its ATM10 the last update 1.26 released with 2.5.9 (not trying to be patronizing, I'm also dealing with the same issue on that pack)

**Comment by Erelaszun** [comment]
> Both the client and server are running 2.5.10? If its ATM10 the last update 1.26 released with 2.5.9 (not trying to be patronizing, I'm also dealing with the same issue on that pack)

Yes... I am not running ATM10... It's a personal modpack I am making... And this is a local run, not on a server.

**Comment by stal111** [comment]
If you're still experiencing this issue with the newest version (2.5.10) please use /data get block on the Hephaestus Forge and share the chat output of the command.

**Comment by Erelaszun** [comment]
> If you're still experiencing this issue with the newest version (2.5.10) please use /data get block on the Hephaestus Forge and share the chat output of the command.

Will do as soon as I get off work.

**Comment by Erelaszun** [comment]
[https://imgur.com/a/96vTDRo](https://imgur.com/a/96vTDRo)

Here you can see after its first formed, and after I filled it with Aureal, Souls, Blood, and Experience... and all the pedestals and items needed to upgrade to tier 2 are in place, but no animation as normal, and clicking with gavel just gives sound effect, but nothing happens

**Comment by stal111** [comment]
> https://imgur.com/a/96vTDRo
> 
> Here you can see after its first formed, and after I filled it with Aureal, Souls, Blood, and Experience... and all the pedestals and items needed to upgrade to tier 2 are in place, but no animation as normal, and clicking with gavel just gives sound effect, but nothing happens

The forge seems to not recognize the items placed on the pedestals (see the empty ingredients array in the data_cache section). Does replacing all the pedestals help?

**Comment by Erelaszun** [comment]
nope... and it doesn't matter if I create them normal, via creative, or place down magnetic... doesn't help at all

Just tested the latest version... no change.

**Comment by FResetFrisk** [comment]
latest version of atm10 pack and forbidden the forge doesnt work. tested in the atm10 public server and a personal private server. please fix these seeming server related issues

**Comment by r-mc2** [comment]
Some observations I've made with a player on my server who isn't able to get rituals to work either:
* The `/data` command output is telling me that the Forge block is registering pedestals that no longer exist
* This player tried creating the Forge in one location and later moved it
* The `/data` command is spitting out pedestals at the old location that have been removed _and_ it's still registering the items that were on those pedestals when they did exist

It would seem the Forge block itself is caching pedestal location/inventory info when it should not be & this seems to be the reason a recipe will not register as valid.

If I'm not mis-reading the JSON output here, that's 12 items in the ingredients list?
![image](https://github.com/user-attachments/assets/5513c9bb-8fb0-4a93-9e22-c24a813a1b9a)

13 if you include the Quantum Core in the Forge block itself.


(Edit: Should note we were trying to craft the Quantum Catcher when the `/data` output was generated. 8 pedestals and only 4 spawner scrap as per the recipe)

Edit 2:
I have since gone back to my own base and tried using my Forge with no luck. Running the `/data` command my Forge block is now registering the other players pedestals as well as my own so now I have too many ingredients to accomplish any recipes.

**Comment by StarSunder01** [comment]
Same issue, on multiplayer server.  Fix when ?
